packages/amplify-cli-core/src/amplify-node-pkg-detector/package-lock-parser.ts

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash';
+import { AmplifyFault } from '../errors/amplify-fault';
+import { JSONUtilities } from '../jsonUtilities';
+import { LockfileType } from './lock-file-types';
+
+/**
+ * package lock interface
+ */
+export interface PackageLock {
+    name: string;
+    version: string;
+    dependencies?: PackageLockDependency;
+    lockfileVersion: 1 | 2;
+    type: LockfileType.NPM;
+}
+
+/**
+ * package lock dependency interface
+ */
+export interface PackageLockDependency {
+    [depName: string]: PackageLockDependencyType;
+}
+
+/**
+ * package lock dependencies type
+ */
+export interface PackageLockDependencyType {
+    version: string;
+    requires?: {
+      [depName: string]: string;
+    };
+    dependencies?: PackageLockDependency;
+    dev?: boolean;
+}
+
+/**
+ * package lock parser
+ */
+export class PackageLockParser {
```

Comment:
Should this class have `NPM` in it to complement `YarnLockParser`

Reply:
Its present in constructor

Reply:
Sorry I meant in the name of the class.